tasks/broad/UltimaGenomicsGermlineFilteringThreshold.wdl

```diff
@@ -0,0 +1,388 @@
+    Int disk_size
+    String docker = "gcr.io/terra-project-249020/jukebox_vc:test_jc_optimize_3d7509"
+  }
+
+  String used_flow_order = (if flow_order=="" then "TACG" else flow_order)
+
+  command <<<
+    set -e
+
+    source ~/.bashrc
+    conda activate genomics.py3
+
+    python -m tarfile -e ~{ref_fasta_sdf} ~{ref_fasta}.sdf
+
+    run_comparison_pipeline.py \
+            --n_parts 0 \
+            --hpol_filter_length_dist 12 10 \
+            --input_prefix $(echo "~{input_vcf}" | sed 's/\(.vcf.gz\|.vcf\)$//') \
+            --output_file ~{input_vcf_name}.comp.h5 \
+            --gtr_vcf ~{gtr_vcf} \
+            --highconf_intervals ~{gtr_highconf} \
+            --runs_intervals ~{runs_file} \
+            --reference ~{ref_fasta} \
+            --reference_dict ~{ref_dict} \
+            --call_sample_name ~{left_sample_name} \
+            --ignore_filter_status \
+            --flow_order ~{used_flow_order} \
+            --truth_sample_name ~{right_sample_name}\
+            --annotate_intervals ~{sep=" --annotate_intervals " annotation_intervals} \
+            --n_jobs 8 \
+            --output_suffix '' \
+            --output_interval interval_~{input_vcf_name}.comp.bed
+    >>>
+  runtime {
+    memory: "32 GB"
+    cpu: 16
+    disks: "local-disk " + disk_size + " SSD"
+    docker: docker
+  }
+  output {
+    File compare_h5 = "~{input_vcf_name}.comp.h5"
+    Array[File] comparison_beds = glob("~{input_vcf_name}.*.bed")
+    File output_interval = "interval_~{input_vcf_name}.comp.bed"
+  }
+}
+
+task EvaluateResults {
+  input {
+    File h5_input
+    String input_vcf_name
+    Int disk_size
+    String docker = "gcr.io/terra-project-249020/jukebox_vc:test_jc_optimize_3d7509"
+    String score_key
+  }
+  command <<<
+    set -e
+
+    source ~/.bashrc
+    conda activate genomics.py3
+
+    evaluate_concordance.py \
+            --input_file ~{h5_input} \
+            --output_prefix ~{input_vcf_name}.report \
+            --use_for_group_testing variant_type \
+            --score_key ~{score_key}
+
+    >>>
+  runtime {
+    memory: "32 GB"
+    disks: "local-disk " + disk_size + " HDD"
+    docker: docker
+  }
+  output {
+    File eval_report_h5 = "~{input_vcf_name}.report.h5"
+    File thresholds_report = "~{input_vcf_name}.report.thresholds.txt"
+  }
+}
+
+task HardThresholdVCF {
+  input { 
+    File thresholds
+    File input_vcf
+    File input_vcf_index
+    String output_basename
+    String score_key
+    Int disk_size
+    String docker = "us.gcr.io/broad-dsde-methods/broad-gatk-snapshots/gatk-remote-builds:mshand-05a76349aac401cc5d08ad0375c45fa9d4c7c864-4.2.6.1-48-g05a76349a"
+  }
+
+  command <<<
+    set -eo pipefail
+    
+    tail -n +2 ~{thresholds} > ~{thresholds}.body
+    cat ~{thresholds}.body | awk 'BEGIN { FS=","} \
+    {print "-filter \"VARIANT_TYPE == \047" $1 "\047 && ~{score_key} \
+     < "$2 "\" --filter-name LOW_SCORE_"$1 " "}' | tr '\n' ' ' > tmpfile
```

Comment:
I can't decide if this is fragile or not.  What leapt out at me is how long I was staring at this, trying to unravel how this works.  I don't think this needs to be rewritten, but in case it does become fragile, I suggest adding a comment or two to help unpack its intent.

Reply:
Yes, let's comment and perhaps simplify. I don't think I've unraveled it myself!

For example, not sure I understand the presence of `LOW_SCORE_`. Is this trying to construct the filter name used by ScoreVariantAnnotations? If so, note that the filter name there is also an exposed argument, and it's not necessarily given by `LOW_` + `score_key`; it's just that the default key and filter names are `SCORE` and `LOW_SCORE`, respectively.

Reply:
I attempted a rewrite of this that will hopefully be clearer (with an additional comment too). 

The `LOW_SCORE_` is trying to construct a filter name of the type of variant (it's broken up by snp, hmer indel, nonhmer indel, etc.) which is not clear at all unless you know what's in the `thresholds` file. Maybe it makes more sense to name the filter `"LOW_" + name_of_score_key + "_" + variant_type`, but maybe it's cleaner to hard code the filter name as `"LOW_SCORE_" + variant_type`? It's convenient/confusing that the annotation happens to be `SCORE` in this case.

Either way, I'm testing it now and will push the changes on Monday once it works.